src/Compilers/Core/MSBuildTask/Microsoft.Managed.Core.targets

```diff
@@ -50,6 +50,28 @@
     </PropertyGroup>
   </Target>
 
+  <Target Name="_ComputeSkipAnalyzers" BeforeTargets="CoreCompile">
+    <!-- First, force clear non-user facing property '_SkipAnalyzers'. -->
+    <PropertyGroup>
+      <_SkipAnalyzers></_SkipAnalyzers>
+    </PropertyGroup>
+
+    <!--
+        Then, determine if '_SkipAnalyzers' needs to be 'true' based on user facing property 'RunAnalyzers'.
+        If 'RunAnalyzers' is not set, then fallback to user facing property 'RunAnalyzersDuringBuild'.
+        Latter property allows users to disable analyzers only for non-design time builds.
+        Design time builds are background builds inside Visual Studio,
+        see details here: https://github.com/dotnet/project-system/blob/master/docs/design-time-builds.md.
+        Setting 'RunAnalyzersDuringBuild' to false, without setting 'RunAnalyzers', allows users to
+        continue running analyzers in the background in Visual Studio while typing (i.e. intellisense),
+        while disabling analyzer execution on explicitly invoked non-design time builds.
+      -->
+    <PropertyGroup Condition="'$(RunAnalyzers)' == 'false' or
+                              ('$(RunAnalyzers)' == '' and '$(RunAnalyzersDuringBuild)' == 'false')">
```

Comment:
Who, or what, would be expected to define `$(RunAnalyzersDuringBuild)`? Is that a property that we are setting today in the tasks / targets that we own?

Reply:
This is set by the project system when user clicks on the "Run on build" checkbox here: https://docs.microsoft.com/en-us/visualstudio/code-quality/disable-code-analysis?view=vs-2019#net-core-and-net-standard-projects

Reply:
More specifically: https://github.com/dotnet/project-system/blob/ad6afdd4f59e434f7624127fa5a6993d48641e15/src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Rules/GeneralBrowseObject.xaml#L412-L420